### PR TITLE
[LEGACY] Add a collision fly mode

### PR DESCRIPTION
**Pull request by gabrielvicenteYT**
Nothing else to say really, just a really nice addition…

### CODE REVIEW

**Comment by CzechHek** [comment]
You should replace all Blocks.air and such at this point.

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/flymodes/other/Collide.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package net.ccbluex.liquidbounce.features.module.modules.movement.flymodes.other
+
+import net.ccbluex.liquidbounce.event.BlockBBEvent
+import net.ccbluex.liquidbounce.features.module.modules.movement.flymodes.FlyMode
+import net.minecraft.util.AxisAlignedBB
+
+object Collide : FlyMode("Collide") {
+    override fun onBB(event: BlockBBEvent) {
+        event.boundingBox = AxisAlignedBB(
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/flymodes/other/Collide.kt]: This is weird...

(the following lines)

Comment (gabrielvicenteYT) [reply src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/flymodes/other/Collide.kt]: It's the best I could do, like converting old code to new code, yeah...

**Comment by gabrielvicenteYT** [comment]
> You should replace all Blocks.air and such at this point.

I'll have to see if the class / object has things that check for other blocks.

**Comment by gabrielvicenteYT** [comment]
Most of the things are done.

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/flymodes/other/Collide.kt (hunk 1 above)
Comment (gabrielvicenteYT) [reply src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/flymodes/other/Collide.kt]: Elaborate

Comment (CzechHek) [reply src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/flymodes/other/Collide.kt]: You usually offset by -2, 2, ... and so on, not mentioning that the offset makes it unclear at first sight.

Comment (gabrielvicenteYT) [reply src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/flymodes/other/Collide.kt]: Yeah, but it works… I'll probably make a pull request surrounding those issues.

**Pull request merged**